File: app/src/main/java/io/dwak/kotlinsample/data/NoteDao.java
package io.dwak.kotlinsample.data;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

@Dao
public interface NoteDao {
    @Query("SELECT * FROM note")
    LiveData<List<Note>> allNotes();

    @Query("SELECT * FROM note WHERE id = :id")
    LiveData<Note> note(String id);

    @Insert
    void insertAll(Note... notes);

    @Update
    void update(Note... note);

    @Delete
    void delete(Note note);
}

File: app/src/main/java/io/dwak/kotlinsample/NoteDao.kt
package io.dwak.kotlinsample

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Delete
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query

@Dao
interface NoteDao {
  @Query("SELECT * FROM note")
  fun allNotes(): LiveData<List<Note>>

  @Query("SELECT * FROM note WHERE id = :arg0")
  fun note(id: String): LiveData<Note>

  @Insert
  fun insertAll(vararg note: Note)

  @Delete
  fun delete(note: Note)
}